multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -254,7 +304,19 @@ def __init__(
             without its weights.
         sample_data_path
             This is used for automatically inference num_classes, classes, or label.
+
         """
+        # Parse problem_type to pipeline
+        if problem_type in PROBLEM_TYPES_SUPPORT_INFERENCE:
+            if pipeline is None:
+                pipeline = problem_type
+            else:
+                assert pipeline == problem_type, (
```

Comment:
I think this causes some inconsistency between the `fit()` and `init()` APIs. The presets need to be passed to `problem_type` in `init()`, but to `presets` in `fit()`.

Reply:
We should submit another PR to support `presets` in `__init__()`. The current logic of using presets as pipeline keys in matching should be refactored.

Currently, I'm keeping the `pipeline` design. `pipeline` essentially means that we can call `predictor.predict()` and `predictor.evaluate()` without calling `predictor.fit()`.

Problem type means the predictor support `predictor.fit()`.

Reply:
I think the current pipeline has the same semantic meaning as `presets`. I guess simply replacing `pipeline` with `presets` should work. No need to keep both `presets` and `pipeline`.